mypy.ini

```diff
@@ -63,6 +63,9 @@ ignore_missing_imports = True
 [mypy-pythran]
 ignore_missing_imports = True
 
+[mypy-mpmath]
+ignore_missing_imports = True
+
```

Comment:
```suggestion
```

I wanted to do this is gh-13613 but a lot of places in SciPy have the `type: ignore[import]` directive where `mpmath` is imported. Doing that already indicates `mypy` to skip the import error we see on master. So, this change will conflict with those directives and `mypy` will mark them unused. Those are the failures in this PR. So, getting rid of these lines should resolve all the errors.

Another solution is to remove the `type: ignore[import]` directive wherever `mpmath` is imported in SciPy and then ignore it in the `mypy.ini` file. This will help avoid failures in the future when `mpmath` is imported without the need to use the directive ever again. But it can be done in a separate PR. Do you agree?

Reply:
> Another solution is to remove the `type: ignore[import]` directive wherever `mpmath` is imported in SciPy and then add these lines in the `mypy.ini` file. This will help avoid failures in the future when `mpmath` is imported without the need to use the directive ever again. But it can be done in a separate PR. Do you agree?

Yeah, I did a search and looks like 14 entries across our code. 
Unless anyone objects I would prefer having the single config change instead of the inlines.